### PR TITLE
expanded alert component to handle auto hide

### PR DESCRIPTION
**Pull request by joefitter**

### CODE REVIEW

**Review by wheelsandcogs — APPROVED** [review]
Looks ok.

**Comment by joefitter** [comment]
WIPing this as need a way to pass props to the alert snippet

File: src/notification/index.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  componentDidMount() {
+    this.timer = this.timer.bind(this);
+    this.timer();
+  }
+
+  timer() {
+    if (this.props.timeout && this.props.message) {
+      notificationTimeout = setTimeout(this.props.hideNotification, this.props.timeout);
+    }
```
Comment (lennym) [inline src/notification/index.jsx:21]: We probably want to clear the timeout somewhere around here too.

In the case where a notification fires, and then 4 seconds later another one fires, the latter would only be displayed for 1 second.

File: src/wrapper/index.jsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import React, { Fragment } from 'react';
-import { Alert } from '../';
+import { Notifiction } from '../';
```
Comment (lennym) [inline src/wrapper/index.jsx]: Typo.

**Review by lennym — CHANGES_REQUESTED** [review]

File: src/wrapper/index.jsx (hunk 2 above)
Comment (lennym) [reply src/wrapper/index.jsx]: It's mildly alarming that this passed CI.

File: src/notification/index.jsx (hunk 1 above)
Comment (joefitter) [reply src/notification/index.jsx:21]: yeah good point

File: src/wrapper/index.jsx (hunk 2 above)
Comment (joefitter) [reply src/wrapper/index.jsx]: I'm guessing it must either be `undefined` by default (which is CRAZY) or more likely: rollup sucks

Comment (joefitter) [reply src/wrapper/index.jsx]: another reason we should get rid of the build - no errors are currently thrown on compile

File: src/notification/index.jsx (hunk 1 above)
Comment (lennym) [reply src/notification/index.jsx:21]: I'd also move it outside the condition, to cover even more extreme edge cases where the first notification was se to time out, but the second was not.

**Review by lennym — APPROVED** [review]

**Pull request merged**